### PR TITLE
Entity properties - new draggable numbers

### PR DESCRIPTION
**Pull request by dback2**
Replace number and slider inputs in the entity properties window with a draggable number field per Alexia's design.

Addresses:
https://highfidelity.manuscript.com/f/cases/19062

Test Plan:
- Enter Interface in desktop mode into any domain
- Open Create app and select or create any entity and view the Properties tab
- Verify the step up/down arrow buttons on the right side of any property number field are now replaced with left and right arrows on each side
- Verify when hovering over the number text (but not over the arrow buttons) that a left/right resize cursor appears
- Verify if holding mouse down on the number text itself (but not over the arrow buttons) and dragging left before releasing mouse that it decreases the number and if dragging right that it increases the number
- Verify if clicking on the number text itself (but not over the arrow buttons) and letting go without moving the mouse that the number field maintains the same value and switches to a state with no step arrows, the text is highlighted, and the text can be edited via keyboard. Verify once clicking away from the number field that it returns to the previous state with arrows and that the number is still exactly the same value.
- Verify all forms of number fields are replaced with this new draggable number and perform as expected which includes regular numbers such as Parent Joint Index and Lifespan, sliders such as particle Size/Alpha/Spin properties, vectors such as Position, Rotation, and Dimension

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15022/HighFidelity-Beta-PR14382-4eab623.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15022/HighFidelity-Beta-PR14382-4eab623.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15022/HighFidelity-Beta-Interface-PR14382-4eab623.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15022/HighFidelity-Beta-PR14382-4eab623.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15022/HighFidelity-Beta-Interface-PR14382-4eab623.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15038/HighFidelity-Beta-PR14382-bdded0d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15038/HighFidelity-Beta-PR14382-bdded0d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15038/HighFidelity-Beta-Interface-PR14382-bdded0d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15038/HighFidelity-Beta-PR14382-bdded0d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15038/HighFidelity-Beta-Interface-PR14382-bdded0d.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15060/HighFidelity-Beta-PR14382-b3ccce9.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15060/HighFidelity-Beta-PR14382-b3ccce9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15060/HighFidelity-Beta-Interface-PR14382-b3ccce9.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15107/HighFidelity-Beta-PR14382-91db81c.apk

File: scripts/system/html/js/draggableNumber.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        that.mouseDown(event);
+    },
+    
+    mouseDown: function(event) {
+        this.startEvent = event;
+        document.addEventListener("mousemove", this.onMouseMove);
+        document.addEventListener("mouseup", this.onMouseUp);
+    },
+    
+    mouseMove: function(event, that) {
```
Comment (huffman) [inline scripts/system/html/js/draggableNumber.js]: What is `that` for?

File: scripts/system/html/js/draggableNumber.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        if (this.startEvent) {
+            let dx = event.clientX - this.startEvent.clientX;
+            let valueChange = dx * this.step;
+            let newValue = parseFloat(this.elInput.value) + valueChange;
```
Comment (huffman) [inline scripts/system/html/js/draggableNumber.js]: I thought we used the input's stepping function here?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15107/HighFidelity-Beta-PR14382-91db81c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15107/HighFidelity-Beta-Interface-PR14382-91db81c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15107/HighFidelity-Beta-PR14382-91db81c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15107/HighFidelity-Beta-Interface-PR14382-91db81c.dmg

File: scripts/system/html/js/draggableNumber.js (hunk 2 above)
Comment (dback2) [reply scripts/system/html/js/draggableNumber.js]: We do, this is now updated

File: scripts/system/html/js/draggableNumber.js (hunk 1 above)
Comment (dback2) [reply scripts/system/html/js/draggableNumber.js]: It was getting around the scope that the callbacks were on the element's scope vs the DraggableNumber scope, but latest update fixes that with bindings.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15117/HighFidelity-Beta-PR14382-8db81d9.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15117/HighFidelity-Beta-PR14382-8db81d9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15117/HighFidelity-Beta-Interface-PR14382-8db81d9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15117/HighFidelity-Beta-PR14382-8db81d9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15117/HighFidelity-Beta-Interface-PR14382-8db81d9.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15131/HighFidelity-Beta-PR14382-1aa048f.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15131/HighFidelity-Beta-PR14382-1aa048f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15131/HighFidelity-Beta-Interface-PR14382-1aa048f.exe

**Comment by jakram** [comment]
@conklin94122 FAIL

2 issues:

1. Physics settings - Angular dampening, Bounciness, Friction, and Density - these settings - snap back to 0 when choosing a negative number, using the slider cursor. However, when using arrow button to choose a negative number, I'm able to do so.

2. Changing the Gravity settings, by using the slider cursor, I can change the numbers of all three axis by dragging the cursor off the Create app and onto the interface.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15169/HighFidelity-Beta-PR14382-d8f5e0f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15169/HighFidelity-Beta-PR14382-d8f5e0f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15169/HighFidelity-Beta-Interface-PR14382-d8f5e0f.dmg

**Comment by dback2** [comment]
@jakram I have fixed the issues in 1, but for 2 that is expected behavior that the other gravity axis values can change when changing one axis, and you can see this on master if setting different huge values to each axis manually. Basically we are trying to maintain a limit on the overall length of the vector, so all 3 values may get clamped when 1 value changes to allow for that. This same behavior can also occur with the Linear Velocity and Angular Velocity vectors, though their vector length limits are higher than Gravity. With the new draggable numbers it is a lot easier to get these limits causing changes in the other values, but for Linear Velocity and Gravity I have decreased the incremental step for each value change from 1 to 0.1 to help avoid this.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15176/HighFidelity-Beta-PR14382-dfed797.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15176/HighFidelity-Beta-PR14382-dfed797.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15176/HighFidelity-Beta-Interface-PR14382-dfed797.dmg

**Pull request merged**